manager/manager/types.go

```diff
@@ -140,6 +140,23 @@ type Configuration struct {
 	Resources []*Resource `json:"resources"`
 }
 
+// ResourceStatus is an enumeration type for the status of a resource.
+type ResourceStatus string
+
+// These constants implement the resourceStatus enumeration type.
+const (
+	Created ResourceStatus = "Created"
+	Failed  ResourceStatus = "Failed"
+	Aborted ResourceStatus = "Aborted"
+)
+
+// ResourceState describes the state of a resource.
+type ResourceState struct {
+	Status   ResourceStatus `json:"status"`
```

Comment:
Yes, there's always one of the states there. Added omitempty and a comment that it's a terminal state. PTAL.